### ISSUE TITLE
filter_oplog_entry not filtering fields when operation is 'u'

### ISSUE THREAD

**Issue by sjalali** [mongodb-labs/mongo-connector#256]
During initial dump, fields is honored because they are used in a Mongo projection.
For an insert, fields are handled correctly.
For an update, $set and $unset are both always {} so nothing is done in filter_oplog_entry.

2015-05-20 16:40:32,012 [DEBUG] mongo_connector.oplog_manager:349 - $set={}, $unset={}
2015-05-20 16:40:32,012 [DEBUG] mongo_connector.oplog_manager:235 - OplogThread: Operation for this entry is u

**Comment by llvtt** [comment]
Can you provide an example of a document that you're trying to update and the update operation that doesn't seem to respect `--fields`?

**Comment by sjalali** [comment]
http://pastie.org/private/0iwjhg732tzkd3xzwutg

**Comment by llvtt** [comment]
@sjalali Thanks for the info. I notice that Elasticsearch is responding with status `400` to Mongo Connector's update requests. Could you also provide logs from Elasticsearch from during one of these update operations?

**Comment by sjalali** [comment]
The ES error is at the bottom of this pastie:
http://pastie.org/private/r255wc58yms2hxilqvsaa
Note that the data in error should not have reached ES but should have been excluded by the fields array.

**Comment by sjalali** [comment]
@llvtt Hi, do you need any more info from me?

**Comment by llvtt** [comment]
@sjalali I think I have all the information I need. Thank you.

**Comment by llvtt** [comment]
@sjalali
Thanks for reporting this bug. It's been fixed in master, and in the process I've also discovered another shortcoming of `--fields` of which I wasn't previously aware. I'm not sure if this limitation affects you or not, since your problem seems to be rooted in how mongo-connector handles document replacement.

The limitation: `--fields` doesn't fully support filtering out nested fields (i.e., fields in subdocuments). It *happens* to work when doing inserts and updates; the only case I've found where the filtering doesn't work is when replacing a document with another one (i.e., a `ReplaceOne` write operation). Some work will be required to implement this, so for now I'm going to document this limitation, and I've added a warning message to print if a user tries to run mongo-connector passing in a nested field to `--fields` (mongo-connector will still proceed if you want it to). 

I've modified issue #255 to be a feature request to support filtering on nested fields fully.

Thanks for all the patience and info you've provided. Let me know if you're still having trouble.
